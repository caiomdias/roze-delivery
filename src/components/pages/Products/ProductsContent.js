import React, { useEffect, useState } from 'react';
import Shelf from './Shelf';
import { getCategories } from '../../../services/ZeApi'

const ProductsContent = ({ distributorId }) =>  {
  const [ allCategories, setAllCategories ] = useState([])
  const [ distributorIdState, setDistributorIdState ] = useState(distributorId || '')

  const variables = {
    id: distributorIdState,
    search: '',
    categoryId: null
  }

  useEffect(() => {
    getCategories(variables)
      .then(({ data }) => {
        if(data.allCategory.length != 0) {
          setAllCategories(data.allCategory)
        }
      })
    return;
  }, [])

  return (
    <div className="products-content">
      <div className="container">
        {
          allCategories
          ? allCategories.map((item, index) =>
            (
            <div key={index} className="item">
              <p>{item.title}</p>
              <Shelf categoryId={item.id} distributorId={distributorIdState}/>
            </div>
            ))
          : ''
        }
      </div>
    </div>
  );
  };

ProductsContent.defaultProps = {
  distributorId: ''
}

export default ProductsContent;
