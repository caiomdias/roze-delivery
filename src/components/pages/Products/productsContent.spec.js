import React from 'react';
import renderer from 'react-test-renderer';
import ProductsContent from './ProductsContent';

jest.mock('react-router-dom', () => ({
    useLocation: jest.fn().mockReturnValue({
      pathname: '/',
      search: '',
      hash: '',
      state: null,
      key: '5nvxpbdafa',
	}),
	useHistory: jest.fn().mockReturnValue({
		push: jest.fn().mockReturnValue()
	}),
}));

const props = {
	distributorId: '111'
}

describe("Products Content Component", () => {
	it('Should render correctly ProductsContent component', () => {  
		const productsContentComponent = renderer.create(<ProductsContent {...props}/>);
		expect(productsContentComponent).toMatchSnapshot();
	});
});