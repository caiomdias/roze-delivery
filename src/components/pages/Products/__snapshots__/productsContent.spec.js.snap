// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products Content Component Should render correctly ProductsContent component 1`] = `
<div
  className="products-content"
>
  <div
    className="container"
  />
</div>
`;
